lib/index.js

```diff
@@ -38,6 +42,35 @@ function initialize(config) {
   });
 }
 
+function filterPreviousUploads(uploadListPath, objects) {
+  var uploadedFileList;
+  if (uploadListPath) {
+    uploadedFileList = fs.readFileSync(uploadListPath, {encoding: 'utf8'});
+    uploadedFileList = uploadedFileList.split('\n');
+    if (!uploadedFileList[uploadedFileList.length-1]) uploadedFileList.pop();
+
+    uploadedFileList = _.map(uploadedFileList, function(anUploadedFile) {
+      return JSON.parse(anUploadedFile);
+    });
+
+    uploadedFilesById = _.groupBy(uploadedFileList, 'objectId');
+  }
+
+  objects = _.filter(objects, function(aFileObject) {
+    if (!aFileObject.fileName) return false; //null fileName causes error in indexOf later
```

Comment:
So basically, when the script gets all the objects w/file it does a query where is checks if the fields exist. The ones with null file names get returned in the results. Later, when it post processes the results for form object like this, but only adds `fileName` and `url` if `fileName` is not null and not undefined. So there's sort of a check, but it doesn't remove the objects that will cause indexOf to error later:
```
var returnValue = {
            className: schema.className,
            objectId: result.id,
            fieldName: field,
            fileName = fileName, //only added if fileName is not null or undefined
            url = url //only added if fileName is not null or undefined
          }
```

Theoretically we could filter the array of the above objects and remove them right after this (inside `getObjectsWithFilesFromSchema`) but that is basically what we're already doing by calling another method from `initialize`

At first I thought it would make sense to do this validation in `_isParseHostedFile` where the indexOf but that's a boolean and has no real way of smoothly skipping an object. And it's called in a few places.